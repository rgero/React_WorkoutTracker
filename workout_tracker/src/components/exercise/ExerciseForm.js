import React from 'react';
import Alert from 'react-bootstrap/Alert';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Form from 'react-bootstrap/Form';
import InputGroup from 'react-bootstrap/InputGroup';

import { SetForm } from '../set/SetForm';
import { SetList } from '../set/SetList';

export const ExerciseForm = ({exercise = {}, onSubmit})=> 
{
    const [setList, setSetList] = React.useState(exercise.setList ? exercise.setList : []);
    const [muscleGroup, setMuscleGroup] = React.useState(exercise.muscleGroup ? exercise.muscleGroup : "");
    const [name, setExerciseName] = React.useState(exercise.name ? exercise.name : "");
    const [notes, setNotes] = React.useState(exercise.notes ? exercise.notes : "");
    const [errMsg, setError] = React.useState("");

    const addSet = (set) => {
        setSetList([...setList, set]);
    }
    
    const deleteSet = (index) => {
        let newSetList = setList;
        newSetList.splice(index, 1);
        setSetList( [...newSetList] );
    }

    const processExercise = (e) => {
        e.preventDefault();

        if (name === "")
        {
            setError("Missing exercise name");
            return;
        }

        if (setList.length === 0)
        {
            setError("Missing set list");
            return;
        }

        const newExercise = {
            name,
            muscleGroup,
            setList,
            notes
        };

        onSubmit(newExercise);
        setSetList([]);
        setMuscleGroup("")
        setExerciseName("");
        setNotes("");
        setError("");
    }

    return (
        <Container fluid="md">
            <Form id="addExerciseForm" onSubmit={processExercise}>
                <InputGroup className="mb-3">
                    <InputGroup.Text>Name</InputGroup.Text>
                    <Form.Control
                        aria-label="name" 
                        type="text" 
                        value={name} 
                        onChange={e => setExerciseName(e.target.value)}
                    />
                </InputGroup>
                <InputGroup className="mb-3">
                    <InputGroup.Text>Muscle Group</InputGroup.Text>
                    <Form.Control
                        aria-label="muscleGroup"
                        type="text"
                        value={muscleGroup}
                        onChange={e => setMuscleGroup(e.target.value)}
                    />
                </InputGroup>
                <InputGroup className="mb-3">
                    <InputGroup.Text>Notes</InputGroup.Text>
                    <Form.Control
                        aria-label="exerciseNotes"
                        as="textarea"
                        value={notes}
                        rows={3}
                        onChange={e => setNotes(e.target.value)}
                    />
                </InputGroup>
            </Form>
            <SetForm onSubmit={addSet}/>
            <SetList setList={setList} onDelete={deleteSet}/>
            <Button
                aria-label="exerciseSubmit" 
                className="mt-4" 
                form="addExerciseForm" 
                variant="outline-secondary" 
                type="submit"
            >
                    Add Exercise
            </Button>

            {/* Error Processing? Don't know if this is completely necessary */}
            { errMsg ? (
                <Container className="pt-4">
                    <Alert key='danger' variant='danger'>
                        Error: {errMsg}
                    </Alert>
                </Container>
            ): ( null )}
        </Container>
    )
}