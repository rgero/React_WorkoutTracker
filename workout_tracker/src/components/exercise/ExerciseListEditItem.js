import React from 'react';

import { SetList } from '../set/SetList';
import {AlertBox} from '../../helpers/DialogBox';

import Container from 'react-bootstrap/Container';
import Card from 'react-bootstrap/Card';

export const ExerciseListItem = ({index, exercise = {}, onDelete})=> 
{
  const processDelete = () => {
    let title = 'Are you sure you want to delete this?';
    let subtitle = `The exercise named ${exercise.name}`;
    let buttons = [
      {
        label: 'Yes',
        onClick: async () => {
          await onDelete(exercise);
        }
      },
      {
        label: 'No'
      }
    ]
    AlertBox(title, subtitle, buttons);
  }

  return (
    <>
      {
        Object.keys(exercise).length === 0 ? (
          <Container className="list-item list-item--message">
            <span>No Name</span>
          </Container>
        ) : (
          <Card onClick={processDelete}>
            <Card.Body>
              <Card.Title>#{index} - {exercise.name}</Card.Title>
              <Card.Body>
                <SetList  setList={exercise.setList} key={`${exercise.name}_Sets`}/>
              </Card.Body>
            </Card.Body>
          </Card>
        )
      }
    </>
  )
}