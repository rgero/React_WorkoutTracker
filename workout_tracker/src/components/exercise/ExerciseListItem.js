import React from 'react';

import { SetList } from '../set/SetList';
import {AlertBox} from '../../helpers/DialogBox';

import Container from 'react-bootstrap/Container';

export const ExerciseListItem = ({exercise, onDelete})=> 
{
  let exerciseName = exercise.name;
  let setList = exercise.setList;

  const processDelete = () => {
    let title = 'Are you sure you want to delete this?';
    let subtitle = `The exercise named ${exerciseName}`;
    let buttons = [
      {
        label: 'Yes',
        onClick: async () => {
          await onDelete(exercise);
        }
      },
      {
        label: 'No'
      }
    ]
    AlertBox(title, subtitle, buttons);
  }

  return (
    <>
      {
        exerciseName === "" ? (
          <Container className="list-item list-item--message">
            <span>No Name</span>
          </Container>
        ) : (
            <Container>
                <Container onClick={processDelete}>
                    {exerciseName}
                    <SetList setList={setList} key={`${exerciseName}_Sets`}/>
                </Container>
            </Container>
        )
      }
    </>
  )
}