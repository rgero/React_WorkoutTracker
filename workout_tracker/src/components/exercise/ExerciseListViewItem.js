import React from 'react';

import { SetList } from '../set/SetList';
import {AlertBox} from '../../helpers/DialogBox';

import Container from 'react-bootstrap/Container';
import Card from 'react-bootstrap/Card';

export const ExerciseListItem = ({index, exercise = {}})=> 
{

  return (
    <>
      {
        Object.keys(exercise).length === 0 ? (
          <Container className="list-item list-item--message">
            <span>No Name</span>
          </Container>
        ) : (
          <Card>
            <Card.Body>
              <Card.Title>#{index} - {exercise.name}</Card.Title>
              <Card.Body>
                <SetList  setList={exercise.setList} key={`${exercise.name}_Sets`}/>
              </Card.Body>
            </Card.Body>
          </Card>
        )
      }
    </>
  )
}