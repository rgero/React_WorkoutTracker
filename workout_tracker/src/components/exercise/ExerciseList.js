import React from 'react';
import Container from 'react-bootstrap/Container';
import ListGroup from 'react-bootstrap/ListGroup';

import { ExerciseListItem } from '../exercise/ExerciseListItem';

export const ExerciseList = ({exerciseList = [], onDelete})=> 
{
    return (
        <Container>
            {
                exerciseList.length === 0 ? (
                    <div className="list-item list-item--message">
                        <span>No Exercises</span>
                    </div>
                ) : (
                    <Container>
                        <ListGroup>
                        {
                            exerciseList.map((exercise, index) => (
                                <ListGroup.Item action onClick={(e) => onDelete(index)}>
                                    <ExerciseListItem
                                        exercise={exercise}
                                    />
                                </ListGroup.Item>
                            ))
                        }
                        </ListGroup>

                    </Container>
                )
            }
        </Container>
    )
}