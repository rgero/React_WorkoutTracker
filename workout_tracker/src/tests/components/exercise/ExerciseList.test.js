import TestRenderer from 'react-test-renderer';
import {ExerciseList} from '../../../components/exercise/ExerciseList';

import { TestExercises } from '../../testData';

describe("Render tests", ()=> {

    let mockDelete = jest.fn();

    test("No Data, matches snapshot", ()=> {
        let element = new TestRenderer.create(
            <ExerciseList onDelete={mockDelete}/>
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

    test("Has Data, matches snapshot", ()=> {

        let element = new TestRenderer.create(
            <ExerciseList exerciseList={TestExercises} onDelete={mockDelete}/>
        )
        expect(element.toJSON()).toMatchSnapshot();
    })

})