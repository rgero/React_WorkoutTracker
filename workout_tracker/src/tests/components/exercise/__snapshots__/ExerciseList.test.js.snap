// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests Has Data, matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="list-group list-group-flush"
    onKeyDown={[Function]}
    style={
      Object {
        "margin": "auto",
        "width": "60%",
      }
    }
  >
    <button
      className="list-group-item list-group-item-action"
      data-rr-ui-event-key={null}
      onClick={[Function]}
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            #
            1
             - 
            Lunges
          </div>
          <div
            className="card-body"
          >
            <div
              className="pt-3 container"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                  "width": "60%",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                Set List
              </div>
              <table
                className="table table-sm table-striped table-bordered table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Reps
                    </th>
                    <th>
                      Weight
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </button>
    <button
      className="list-group-item list-group-item-action"
      data-rr-ui-event-key={null}
      onClick={[Function]}
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            #
            2
             - 
            Lunges
          </div>
          <div
            className="card-body"
          >
            <div
              className="pt-3 container"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                  "width": "60%",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                Set List
              </div>
              <table
                className="table table-sm table-striped table-bordered table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Reps
                    </th>
                    <th>
                      Weight
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </button>
    <button
      className="list-group-item list-group-item-action"
      data-rr-ui-event-key={null}
      onClick={[Function]}
      style={
        Object {
          "border": "0px",
        }
      }
    >
      <div
        className="card"
        onClick={[Function]}
      >
        <div
          className="card-body"
        >
          <div
            className="card-title h5"
          >
            #
            3
             - 
            Lunges
          </div>
          <div
            className="card-body"
          >
            <div
              className="pt-3 container"
              style={
                Object {
                  "margin": "auto",
                  "textAlign": "center",
                  "width": "60%",
                }
              }
            >
              <div
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                Set List
              </div>
              <table
                className="table table-sm table-striped table-bordered table-hover"
              >
                <thead>
                  <tr>
                    <th>
                      Reps
                    </th>
                    <th>
                      Weight
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                  <tr
                    onClick={[Function]}
                  >
                    <td>
                      40
                    </td>
                    <td>
                      100
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`Render tests No Data, matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="list-item list-item--message"
  >
    <span>
      No Exercises
    </span>
  </div>
</div>
`;
